sys/ufs/ufs/ufs_lookup.c

```diff
@@ -1394,7 +1394,7 @@ ufs_dir_dd_ino(struct vnode *vp, struct ucred *cred, ino_t *dd_ino,
 	/*
 	 * Have to read the directory.
 	 */
-	error = vn_rdwr(UIO_READ, vp, (caddr_t)&dirbuf,
+	error = vn_rdwr(UIO_READ, vp, &dirbuf,
```

Comment:
PTR2CAP?

Reply:
I assume that this one the compiler generates a capability to directly via the `&` operator.

Reply:
Interestingly it makes `typeof` sad:
```
.../cheri/cheribsd/sys/fs/ext2fs/ext2_lookup.c:1015:55: error: converting capability type 'typeof ((*&dirbuf)) * __capability' (aka 'char ** __capability') to non-capability type 'char **' without an explicit cast; if this is intended use __cheri_fromcap
        if ((error = ext2_blkatoff(dvp, (off_t)dp->i_offset, PTR2CAP(&dirbuf),
                                                             ^
                                                             (__cheri_fromcap char **)
.../cheri/cheribsd/sys/sys/param.h:429:20: note: expanded from macro 'PTR2CAP'
#define PTR2CAP(p)      ({                                      \
                        ^
1 error generated.
```

Reply:
Actually, this case is ok and I've added a PTR2CAP. The once cited above is something else.